### PR TITLE
Remove custom arguments from Maven Release Plugin configuration

### PR DESCRIPTION
**Pull request by basil**
Fixes a regression exposed by jenkinsci/pom#281 and #6914. In order for release artifacts to be signed by `maven-gpg-plugin` and `maven-jarsigner-plugin` during the `release:prepare` goal, we rely on https://github.com/jenkinsci/jenkins/blob/d25704cfbc929bca3bd5e5675e1fe1aa1a58928c/pom.xml#L521-L540 and https://github.com/jenkinsci/jenkins/blob/d25704cfbc929bca3bd5e5675e1fe1aa1a58928c/war/pom.xml#L623-L648, activated by the `release` and `sign` profiles respectively. To activate these profiles during the `release:prepare` goal we currently rely on https://github.com/jenkinsci/jenkins/blob/d25704cfbc929bca3bd5e5675e1fe1aa1a58928c/pom.xml#L428-L429, which happens to expose a serious bug in recent versions of Maven Release Plugin as explained in https://github.com/jenkins-infra/helpdesk/issues/3143#issuecomment-1254040470. Fortunately for us, we can work around the problem by omitting any custom arguments from our Maven Release Plugin configuration as in https://github.com/jenkins-infra/release/pull/291 and this PR, instead enabling these profiles via an `<activeProfiles>` setting in [`settings-release.xml`](https://github.com/jenkins-infra/release/blob/29865baf0d90bf30885f81413bf7dbb36dd36582/utils/release.bash#L152-L237). Note that we cannot enable these profiles using [`releaseProfiles`](https://maven.apache.org/maven-release/maven-release-plugin/stage-mojo.html#releaseProfiles) because that setting only applies to the `release:stage` and `release:perform` goals, not the `release:prepare` goal.

### Proposed changelog entries

Support staging of releases (regression in 2.361).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7138"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Comment by daniel-beck** [comment]
Perhaps beyond the immediate scope of this, but given linked PRs and motivation, could we fail the build if it's a release build, and `-Darguments` is set, or similar? Seems like a year or two from now, nobody will remember this and we'll happily add all sorts of stuff via `-Darguments` again and will have a much more harmful staging experience -- it was pure luck this happened during a really unusual release where we staged not several days before the intended release, but on the same day. (That seems like the cleaner solution, if possible, than adding comments everywhere we're now removing `arguments` from to remember not to do this again.)

**Comment by basil** [comment]
I am hoping the Maven folks will fix their bug, so I see no need to do what you are suggesting.

**Review by daniel-beck — APPROVED** [review]
AFAIUI, depends on https://github.com/jenkins-infra/release/pull/292, which is merged (i.e., this is fine).

**Review by basil — COMMENTED** [review]
I plan to monitor the next weekly release to ensure that signing takes place as expected, then backport this to `stable-2.361`.

---

This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by timja — APPROVED** [review]

**Comment by daniel-beck** [comment]
> I am hoping the Maven folks will fix their bug, so I see no need to do what you are suggesting.

FTR another option might be in core to always require an alternative deployment repository. Even regular release builds (pre-2.361) have

```
02:09:12.888  [INFO] [INFO] --- maven-deploy-plugin:2.8.2:deploy (default-deploy) @ jenkins-parent ---
02:09:12.888  [INFO] [INFO] Using alternate deployment repository releases::default::https://repo.jenkins-ci.org/releases
02:09:12.888  [INFO] [INFO] Uploading to releases: https://repo.jenkins-ci.org/releases/org/jenkins-ci/main/jenkins-parent/2.360/jenkins-parent-2.360.pom
02:09:13.591  [INFO] [INFO] Uploaded to releases: https://repo.jenkins-ci.org/releases/org/jenkins-ci/main/jenkins-parent/2.360/jenkins-parent-2.360.pom (5.2 kB at 7.7 kB/s)

**Comment by basil** [comment]
> I am hoping the Maven folks will fix their bug, so I see no need to do what you are suggesting.

FTR they did, in [3.0.0-M7](https://github.com/apache/maven-release/releases/tag/maven-release-3.0.0-M7).

**Pull request merged**
